test/CadlRanchProjectsNonAzure.Tests/type-property-nullable.cs

```diff
@@ -0,0 +1,248 @@
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsModel_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsModelClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual("hello", response.Value.NullableProperty.First().Property);
+            Assert.AreEqual("world", response.Value.NullableProperty.Last().Property);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsModel_getNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsModelClient().GetNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            // we will never construct a null collection therefore this property is actually undefined here.
+            Assert.IsNotNull(response.Value.NullableProperty);
+            Assert.IsFalse(!(response.Value.NullableProperty is ChangeTrackingList<InnerModel> changeTrackingList && changeTrackingList.IsUndefined));
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsModel_patchNonNull() => Test(async (host) =>
+        {
+            var value = new
+            {
+                requiredProperty = "foo",
+                nullableProperty = new[]
+                {
+                    new
+                    {
+                        property = "hello"
+                    },
+                    new
+                    {
+                        property = "world"
+                    }
+                }
+            };
+            var response = await new NullableClient(host, null).GetCollectionsModelClient().PatchNonNullAsync(BinaryContent.Create(BinaryData.FromObjectAsJson(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsModel_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetCollectionsModelClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
```

Comment:
The test case will fail according to the above: Bad request, nullableProperty cannot be set to null in that way, it can only be set through a string form.